modules/client/src/allowlistVoting/internal/methods.ts

```diff
@@ -219,87 +231,96 @@ export class AllowlistVotingMethods {
-      no: proposalData.no,
-      abstain: proposalData.abstain,
+      startDate: parseInt(proposalData.startDate.toString()),
+      endDate: parseInt(proposalData.endDate.toString()),
+      snapshotBlock: BigInt(proposalData.snapshotBlock.toString()),
+      supportRequired: BigInt(proposalData.supportRequired.toString()),
+      participationRequired: BigInt(
+        proposalData.participationRequired.toString()
+      ),
+      votingPower: BigInt(proposalData.votingPower.toString()),
+      yes: BigInt(proposalData.yes.toString()),
+      no: BigInt(proposalData.no.toString()),
+      abstain: BigInt(proposalData.abstain.toString()),
       actions: proposalData.actions.map(action => ({
         to: action.to,
         data: arrayify(action.data),
-        value: action.value,
+        value: BigInt(action.value.toString()),
       })),
     };
-    return vote;
+    return proposal;
   }
 
   /**
    * Returns what a voter voted on a proposal
    *
-   * @param {BigNumberish} _proposalId
+   * @param {BigInt | number} _proposalId
    * @param {string} _voter
    * @return {*}  {Promise<number>}
    * @memberof AllowlistVotingMethods
    */
-  public getVoteOption(_proposalId: BigNumberish, _voter: string): Promise<number> {
+  public getVoteOption(
+    _proposalId: BigInt | number,
+    _voter: string
+  ): Promise<number> {
     return this.allowlistVoting
       .getPluginInstanceWithSigner()
-      .getVoteOption(_proposalId, _voter);
+      .getVoteOption(BigNumber.from(_proposalId), _voter);
   }
 
   /**
    * Checks if a user is allowed on the given block
    *
    * @param {string} account
-   * @param {BigNumberish} blockNumber
+   * @param {BigInt | number} blockNumber
    * @return {*}  {Promise<boolean>}
    * @memberof AllowlistVotingMethods
    */
   public isAllowed(
     account: string,
-    blockNumber: BigNumberish
+    blockNumber: BigInt | number
   ): Promise<boolean> {
     return this.allowlistVoting
       .getPluginInstanceWithSigner()
-      .isAllowed(account, blockNumber);
+      .isAllowed(account, BigNumber.from(blockNumber));
   }
 
   /**
    * Returns the configured minDuration
    *
-   * @return {*}  {Promise<BigNumber>}
+   * @return {*}  {Promise<number>}
    * @memberof AllowlistVotingMethods
    */
-  public minDuration(): Promise<BigNumber> {
-    return this.allowlistVoting.getPluginInstanceWithSigner().minDuration();
+  public async minDuration(): Promise<number> {
+    return (
+      await this.allowlistVoting.getPluginInstanceWithSigner().minDuration()
+    ).toNumber();
   }
 
   /**
    * Returns the configured participation requirement
    *
-   * @return {*}  {Promise<BigNumber>}
+   * @return {*}  {Promise<number>}
    * @memberof AllowlistVotingMethods
    */
-  public participationRequiredPct(): Promise<BigNumber> {
-    return this.allowlistVoting
-      .getPluginInstanceWithSigner()
-      .participationRequiredPct();
+  public async participationRequiredPct(): Promise<number> {
```

Comment:
Could we name it in terms of the SDK? The contract params are going to be renamed for sure. I suggest `minParticipation` and it should return a number between `0` and `1` as a ratio

Reply:
See also above on `minDuration` and `supportRequiredPct`.... unless there is a layer on top that exports the 0-1 number

Reply:
why a number between 0 and 1 if I may ask?
the contracts cannot return floats so we would need to do extra "math" on the contracts and/or the sdk just to go into this scale. why is 0-100 bad?